src/main/java/com/marklogic/spark/writer/ArbitraryRowConverter.java

```diff
@@ -60,26 +57,49 @@ public Optional<DocBuilder.DocumentInputs> convertRow(InternalRow row) {
             row.setNullAt(this.filePathIndex);
         }
 
-        final String json = convertRowToJSONString(row);
-        AbstractWriteHandle contentHandle = this.xmlRootName != null ?
-            new StringHandle(convertJsonToXml(json)).withFormat(Format.XML) :
-            new StringHandle(json).withFormat(Format.JSON);
+        final String json = this.jsonRowSerializer.serializeRowToJson(row);
 
+        AbstractWriteHandle contentHandle = null;
+        ObjectNode deserializedJson = null;
         ObjectNode uriTemplateValues = null;
-        if (this.uriTemplate != null || this.jsonRootName != null) {
-            ObjectNode jsonObject = readTree(json);
+        final boolean mustRemoveFilePathField = this.filePathIndex > 1 && jsonRowSerializer.isIncludeNullFields();
+
+        if (this.jsonRootName != null || this.xmlRootName != null || this.uriTemplate != null || mustRemoveFilePathField) {
+            deserializedJson = readTree(json);
+            if (mustRemoveFilePathField) {
+                deserializedJson.remove(MARKLOGIC_SPARK_FILE_PATH_COLUMN_NAME);
```

Comment:
Things got a little tricky here. Once you add "ignoreNullFields=true", then the "marklogic_spark_file_path" column gets included in the JSON/XML. And we know a user doesn't want that. So this removes that column after the JSON string is deserialized into an ObjectNode. Added some tests to verify that this works correctly.